pkg/apis/autoscaling/annotation_validation.go

```diff
@@ -115,3 +115,32 @@ func validateMinMaxScale(annotations map[string]string) *apis.FieldError {
 	}
 	return errs
 }
+
+func validateMetric(annotations map[string]string) *apis.FieldError {
+	if metric, ok := annotations[MetricAnnotationKey]; ok {
+		var classValue string
+		if c, ok := annotations[ClassAnnotationKey]; ok {
+			classValue = c
+		} else {
+			// Default to "kpa" class for backward compatibility.
+			classValue = KPA
+		}
+		switch classValue {
+		case KPA:
+			switch metric {
+			case Concurrency, RPS:
+				return nil
+			}
+		case HPA:
+			switch metric {
+			case CPU, Concurrency, RPS:
+				return nil
+			}
+		default:
+			// Leave other classes of PodAutoscaler alone.
+			return nil
+		}
+		return apis.ErrInvalidValue(metric, MetricAnnotationKey)
```

Comment:
so this is unreachable now?

Reply:
No its not unreachable 
When `MetricAnnotationKey` is invalid for class HPA and KPA it will execute that line